File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {

    public int findKthPositive(int[] arr, int k) {
        LinkedList<Integer> missing= new LinkedList<>();
        int current = 1;
        int index = 0;
        while (missing.size() < k) {
            if (index < arr.length && arr[index] == current) ++index; 
            else missing.addLast(current);
            ++current;
        }
        return missing.get(k - 1);
    }
}

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution
{
    public:
        int findKthPositive(vector<int> &arr, int k)
        {
            vector<int> missing;
            int current = 1;
            int index = 0;
            while (missing.size() < k)
            {
                if (index < arr.size() && arr[index] == current)
                    ++index;
                else
                    missing.push_back(current);
                ++current;
            }
            return missing[k - 1];
        }
};